adobe/spectrum-web-components
# build: refactor the doc site build processes to use wireit

## Description
Leverages `wireit` to orchestrate commands within the documentation site project and lowers time to start the docs site from ~30s to ~10s while not even breaking into the main projects scripts (right now type on the packages still seems to be the blocker to further performance wins). On update (even all the way back to the markdown files), we're seeing a similar or in many cases larger benefit from the ability to leverage caching and incremental building to reduce unnecessary work.

- add `wireit` and orchestrate the doc site build
- move the TS build in the docs site form TSC to `esbuild`
- move package prep (build/types/etc) out of the main project and into the wireit cascade
- remove rollup from the watch process which require repositioning the HTML post processing into 11ty

## Related issue(s)
- refs #2735 

## How has this been tested?

    1. Go to the repo on the commandline
    2. Run `yarn docs:start` and see it do awesome things

## Types of changes

## Checklist
